### PR TITLE
[BUGFIX beta] more resilient signal trapped cleanup

### PR DESCRIPTION
**Pull request by stefanpenner**
Cooperative signal handling is complicated, in-order to provide the
experience we want, we take control of this process entirely.

See: ://github.com/sindresorhus/ora/issues/27 for more context

### CODE REVIEW

File: lib/cli/index.js [hunk 1]
```diff
@@ -7,6 +7,10 @@ var logger        = require('heimdalljs-logger')('ember-cli:cli/index');
 var merge         = require('ember-cli-lodash-subset').merge;
 var path          = require('path');
 
+// work around misbehaving libraries, so we can correctly cleanup before
+// actually exiting.
+require('capture-exit').captureExit();
```
Comment (nathanhammond) [inline lib/cli/index.js]: Note why I'm not concerned about this: even if things were to be interrupted before this point it doesn't suffer the negative consequences that we are trying to prevent.

File: lib/models/builder.js [hunk 2]
```diff
@@ -84,6 +86,9 @@ module.exports = Task.extend({
       process.on('SIGINT',  this.onSIGINT.bind(this));
       process.on('SIGTERM', this.onSIGTERM.bind(this));
       process.on('message', this.onMessage.bind(this));
+      exit.onExit(function(code) {
+        return this.cleanupAndExit(code);
```
Comment (nathanhammond) [inline lib/models/builder.js]: `this.cleanup.bind(this);`

File: lib/cli/index.js (hunk 1 above)
Comment (hjdivad) [inline lib/cli/index.js]: @nathanhammond what are you not concerned about?  @Turbo87's suggestion to move this before other requires?

His point seems valid; we want to `captureExit` as soon as possible; this is unrelated to when an interruption occurs because we want to get head not only of interruptions but also people who pluck off `process.exit`

Comment (nathanhammond) [inline lib/cli/index.js]: Didn't think about other picking off `process.exit` in advance. Manual review says that isn't a concern here, but it could accidentally become a problem in the future if not hoisted. Seems good to be addressed.
